File: app/src/main/java/ua/com/foxminded/locationtrackera/model/gps/GpsSource.java
package ua.com.foxminded.locationtrackera.model.gps;

import io.reactivex.rxjava3.core.Observable;
import ua.com.foxminded.locationtrackera.model.locations.UserLocation;

public interface GpsSource {

    void onServiceStarted();

    void setUpServices();

    void startLocationUpdates();

    void registerGpsOrGnssStatusChanges();

    Observable<Integer> getGpsStatusObservable();

    Observable<UserLocation> getLocationObservable();

    void onServiceStopped();

    void onDestroy();
}

File: models/src/main/java/ua/com/foxminded/locationtrackera/models/gps/GpsSource.kt
package ua.com.foxminded.locationtrackera.models.gps

import io.reactivex.rxjava3.core.Observable
import ua.com.foxminded.locationtrackera.models.locations.UserLocation

interface GpsSource {

    val gpsStatusObservable: Observable<Int>

    val locationObservable: Observable<UserLocation>

    fun onServiceStarted()

    fun startLocationUpdates()

    fun registerGpsOrGnssStatusChanges()

    fun onServiceStopped()

    fun onDestroy()
}
